### PR TITLE
Review feedback

### PR DESCRIPTION
**Pull request by tomwilkie**

### CODE REVIEW

File: ipam/space/space_set.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (s *Set) GiveUpSpace() (net.IP, uint32, bool) {
 	}
 
 	if bestStart == nil {
-		utils.Assert(s.NumFreeAddresses() == 0, "Failed to find a range but have free addresses")
+		utils.Assert(s.NumFreeAddresses() == 0)
```
Comment (bboreham) [inline ipam/space/space_set.go]: could use `totalFreeAddresses`

File: ipam/ring/ring.go [hunk 2]
```diff
@@ -483,13 +482,12 @@ func (r *Ring) ReportFree(freespace map[uint32]uint32) {
 			return entries[j].Token >= start
 		})
 
-		utils.Assert(i < len(entries) && entries[i].Token == start &&
-			entries[i].Peer == r.Peername, "Trying to report free on space I don't own")
+		utils.Assert(i < len(entries) && entries[i].Token == start && entries[i].Peer == r.Peername)
```
Comment (bboreham) [inline ipam/ring/ring.go:487]: comment here

File: ipam/ring/ring.go [hunk 3]
```diff
@@ -166,9 +166,9 @@ func (r *Ring) GrantRangeToHost(startIP, endIP net.IP, peer router.PeerName) {
 	})
 	preceedingEntry--
 
-	utils.Assert(len(filteredEntries) > 0, "Cannot grant into an empty ring, as by definition you don't own anything.")
+	utils.Assert(len(filteredEntries) > 0)
 	previousLiveEntry := filteredEntries.entry(preceedingEntry)
-	utils.Assert(previousLiveEntry.Peer == r.Peername, "Trying to grant in a range I don't own")
+	utils.Assert(previousLiveEntry.Peer == r.Peername)
```
Comment (bboreham) [inline ipam/ring/ring.go:172]: comment here

File: ipam/space/space_set.go [hunk 4]
```diff
@@ -53,7 +53,7 @@ func (s *Set) AddSpace(newspace *Space) {
 	defer s.assertInvariants()
 
 	i := s.find(newspace.Start)
-	utils.Assert(i >= len(s.spaces) || !s.spaces[i].Start.Equal(newspace.Start), "inserting space into list already exists!")
+	utils.Assert(i >= len(s.spaces) || !s.spaces[i].Start.Equal(newspace.Start))
```
Comment (bboreham) [inline ipam/space/space_set.go]: `i` can't really be `> len` - that would fail the copy below

**Comment by tomwilkie** [comment]
PTAL

File: ipam/ring/ring.go [hunk 5]
```diff
@@ -483,13 +483,13 @@ func (r *Ring) ReportFree(freespace map[uint32]uint32) {
 			return entries[j].Token >= start
 		})
 
-		utils.Assert(i < len(entries) && entries[i].Token == start &&
-			entries[i].Peer == r.Peername, "Trying to report free on space I don't own")
+		// Trying to report free on space I don't own
```
Comment (bboreham) [inline ipam/ring/ring.go]: This either needs to be a question or a statement of the opposite

**Comment by tomwilkie** [comment]
PTAL

**Pull request merged**